src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -280,6 +280,45 @@ static VehicleInfo_Requests ivi_subrequests[] = {
 }
 #endif  // #ifdef HMI_DBUS_API
 
+smart_objects::SmartObjectSPtr MessageHelper::CreateNotification(
+    mobile_apis::FunctionID::eType function_id, uint32_t app_id) {
+  using smart_objects::SmartObject;
+  LOG4CXX_AUTO_TRACE(logger_);
+  smart_objects::SmartObjectSPtr object(
+      new SmartObject(smart_objects::SmartType_Map));
+  (*object)[strings::params][strings::message_type] =
+      static_cast<int>(kNotification);
+  (*object)[strings::params][strings::function_id] = function_id;
+  (*object)[strings::params][strings::connection_key] = app_id;
+  (*object)[strings::params][strings::protocol_version] =
+      commands::CommandImpl::protocol_version_;
+  (*object)[strings::params][strings::protocol_type] =
+      commands::CommandImpl::mobile_protocol_type_;
+  (*object)[strings::msg_params] =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  return object;
+}
+
+NsSmartDeviceLink::NsSmartObjects::SmartObjectSPtr
+MessageHelper::CreateHMINotification(hmi_apis::FunctionID::eType function_id) {
+  using smart_objects::SmartObject;
+  using smart_objects::SmartObjectSPtr;
+  using smart_objects::SmartType_Map;
+  LOG4CXX_AUTO_TRACE(logger_);
+  SmartObjectSPtr notification_ptr =
+      utils::MakeShared<SmartObject>(SmartType_Map);
+  SmartObject& notification = *notification_ptr;
+  notification[strings::params][strings::message_type] =
+      static_cast<int32_t>(application_manager::MessageType::kNotification);
```

Comment:
why line 290 does not need `application_manager::MessageType::`, but it needs here

Reply:
@yang1070 here we also don't need this
Removed 9a0fa27